docs/release/release-tasks.md

```diff
@@ -310,30 +310,38 @@ The goal of this task to make the book for the current release available under e
    <br>Prior art: [Add release 1.2 book link](https://github.com/kubernetes-sigs/cluster-api/pull/6697)
 
 #### Create PR for release notes
-
+1. Checkout the `main` branch.
+1. Build the release note tools binary.
+   ```bash
+   go build -o bin/notes hack/tools/release/notes.go
+   ```
 1. Checkout the latest commit on the release branch, e.g. `release-1.4`, or the main branch if the release branch doesn't yet exist (e.g. beta release).
-2. Generate release notes with:
+1. Generate release notes with:
 
    ```bash
    # PREVIOUS_TAG should be the last patch release of the previous minor release.
    PREVIOUS_TAG=v1.3.x
    # RELEASE_TAG should be the new desired tag (note: at this point the tag does not yet exist).
-   RELEASE_TAG=v1.4.x make release-notes
+   RELEASE_TAG=v1.4.x
+   # If this is a beta or RC release, add the --pre-release-version flag
+   ./bin/notes --from=$PREVIOUS_TAG > CHANGELOG/${RELEASE_TAG}.md
    ```
 
-3. This will generate a new release notes file at `CHANGELOG/<RELEASE_TAG>.md`. Finalize the release notes:
-    1. Pay close attention to the `## :question: Sort these by hand` section, as it contains items that need to be manually sorted.
-    2. Ensure consistent formatting of entries (e.g. prefix).
-       <br>**Note**: Check against the release notes of the [previous release](https://github.com/kubernetes-sigs/cluster-api/releases/latest), depending on the release branch you are currently working on (e.g. v1.3.6 when working on v1.3.7 or v1.4.2 when working on v1.4.3).
-    3. Merge dependency bump PR entries for the same dependency into a single entry.
-    4. Move minor changes into a single line at the end of each section.
-    5. Sort entries within a section alphabetically.
-    6. Write highlights section based on the initial release notes doc.
-    7. Add the Kubernetes version support section and pay close attention to set the correct versions here.
+1. This will generate a new release notes file at `CHANGELOG/<RELEASE_TAG>.md`. Finalize the release notes:
+    - [ ] Update the `Kubernetes version support section`. If this is a patch release you can most probably copy the same values from the previous patch release notes. Except if this is the release where a new Kubernetes version support is added.
        <br>**Note**: Check our [Kubernetes support policy](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions) in the CAPI book. In case of doubt, reach out to the current release lead.
-    8. **For minor releases** Modify `Changes since v1.x.y` to `Changes since v1.x`
+    - [ ] If this is a `vX.X.0` release, add fill the content for the `Highlights` section. Otherwise, remove the section all together.
+    - [ ] If there a deprecations in this release (for example, a CAPI API version drop), add them, to the `Deprecation Warning` section. Otherwise, remove the section all together.
```

Comment:
```suggestion
    - [ ] If there a deprecations in this release (for example, a CAPI API version drop), add them, to the `Deprecation Warning` section. Otherwise, remove the section altogether.
```